From 4c50a5626930baa5edec350162888654c0006ec5 Mon Sep 17 00:00:00 2001
From: Dongwon Kim <dongwon.kim@intel.com>
Date: Mon, 28 Jul 2025 19:49:55 -0700
Subject: [PATCH 3/3] hw/display/virtio-gpu: Redundant call of
 dpy_gfx_replace_surface

'dpy_gfx_replace_surface' to null surface is done upon reset is done
in virtio_gpu_reset_bh.

Signed-off-by: Dongwon Kim <dongwon.kim@intel.com>
---
 hw/display/virtio-gpu.c | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/hw/display/virtio-gpu.c b/hw/display/virtio-gpu.c
index 76a594973c..7e7f97baae 100644
--- a/hw/display/virtio-gpu.c
+++ b/hw/display/virtio-gpu.c
@@ -1596,7 +1596,6 @@ void virtio_gpu_reset(VirtIODevice *vdev)
 {
     VirtIOGPU *g = VIRTIO_GPU(vdev);
     struct virtio_gpu_ctrl_command *cmd;
-    int i = 0;
 
     if (qemu_in_vcpu_thread()) {
         g->reset_finished = false;
@@ -1621,10 +1620,6 @@ void virtio_gpu_reset(VirtIODevice *vdev)
         g_free(cmd);
     }
 
-    for (i = 0; i < g->parent_obj.conf.max_outputs; i++) {
-        dpy_gfx_replace_surface(g->parent_obj.scanout[i].con, NULL);
-    }
-
     virtio_gpu_base_reset(VIRTIO_GPU_BASE(vdev));
 }
 
-- 
2.43.0

